src/lib/utils/uikitConfigMapper.ts

```diff
@@ -0,0 +1,52 @@
+import { UIKitOptions, CommonUIKitConfigProps } from '../types';
+import { getCaseResolvedReplyType } from './resolvedReplyType';
+
+export function uikitConfigMapper({
+  legacyConfig,
+  uikitOptions = {},
+}: { legacyConfig: CommonUIKitConfigProps, uikitOptions?: UIKitOptions }): UIKitOptions {
+  const {
+    replyType,
+    isMentionEnabled,
+    isReactionEnabled,
+    disableUserProfile,
+    isVoiceMessageEnabled,
+    isTypingIndicatorEnabledOnChannelList,
+    isMessageReceiptStatusEnabledOnChannelList,
+    showSearchIcon,
+  } = legacyConfig;
+  return {
+    common: {
+      enableUsingDefaultUserProfile: uikitOptions.common?.enableUsingDefaultUserProfile
+        ?? typeof disableUserProfile === 'boolean'
+        ? !disableUserProfile
+        : undefined,
```

Comment:
Nit: Will be good to add brackets, or move to fn and add test 

little difficult to read..